UnitTests/UnitTests.UWP/Extensions/Test_RectExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Toolkit.Uwp.Extensions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.Foundation;
+
+namespace UnitTests.Extensions
+{
+    [TestClass]
+    public class Test_RectExtensions
+    {
+        [TestCategory("RectExtensions")]
+        [TestMethod]
+        [DataRow(0, 0, 2, 2, 0, 0, 2, 2, true)]// Full intersection.
+        [DataRow(0, 0, 2, 2, 1, 1, 2, 2, true)]// Partial intersection.
+        [DataRow(0, 0, 2, 2, 2, 0, 2, 2, true)]// Edge intersection.
+        [DataRow(0, 0, 2, 2, 2, 2, 2, 2, true)]// Corner intersection.
+        [DataRow(0, 0, 2, 2, 3, 0, 2, 2, false)]// No intersection.
```

Comment:
This is only testing the right/bottom bounds, we should have some different examples which test more of the top/left bounds as well for completeness.

Reply:
Like the test case method here, seems like a good way to set this up, let's just add a few more scenarios from moving the first rectangle in relation to the 2nd and I think you'll be good, thanks!

Reply:
@michael-hawker Thanks for the suggestion! I added more test cases to test other edges and corners. Sorry so late for the response, I had an interview yesterday, I need to prepare it first.